server.go

```diff
@@ -150,6 +165,48 @@ func (s *Server) serveTLS(ctx context.Context) {
 	}
 }
 
+func (s *Server) serveTCPRouterClients(ctx context.Context) {
+	addr := s.ServerOptions.ClientsAddr()
+	ln, err := net.Listen("tcp", addr)
+	if err != nil {
+		log.Fatalf("failed to start tcp listener at %s: %v ", addr, err)
+	}
+
+	log.Println("started serving tcprouter clients..")
+	for {
+		select {
+		case <-ctx.Done():
+			ln.Close()
+			return
+
+		default:
+			conn, err := ln.Accept()
+			if err != nil {
+				log.Printf("error accepting tcprouter client connection: %v\n", err)
+				continue
+			}
+			// TODO: need to track these connection and close then when needed
+			go s.handleTCPRouterClientConnection(conn)
```

Comment:
No need to create goroutine here, you are just doing the handshake and saving the connection for later. This can happen in this block